src/mbi/relaxed_projection.py

```diff
@@ -0,0 +1,368 @@
+import numpy as np
+import pandas as pd
+import itertools
+from typing import Callable, List, Tuple
+
+import jax
+import jax.numpy as jnp
+import optax
+from .dataset import Dataset
+from .domain import Domain
+
+
+
+#######################################################
+#                                                     #
+#       Here is a the projectable data class          #
+#                                                     #
+#######################################################
+
+class ProjectableData():
```

Comment:
Lets make this a dataclass and have fature_indices_map be a @property (eliminating the __init__ boilerplate)